### PR TITLE
don't regex the topic unless needed

### PR DESCRIPTION
**Pull request by osheroff**
This is worth about 6% of main-thread performance.  Something found while profiling. 

@gabetax @prudhvi @zendesk/rules

### CODE REVIEW

**Comment by prudhvi** [comment]
Good catch 👍  looks safe to me just codewise

File: src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java [hunk 1]
```diff
@@ -97,7 +99,10 @@ private Integer getNumPartitions(String topic) {
 	}
 
 	private String generateTopic(String topic, RowMap r){
```
Comment (prudhvi) [inline src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java]: Also does this method need to take `topic` parameter? Can't it rely on `this.topic`?

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java]: yeah.   but it doesn't matter.  And in many cases anything you do to make something more of a "pure" function is better.

**Comment by smferguson** [comment]
6%?! ugh. thanks for finding/fixing :)

**Pull request merged**